cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -214,7 +215,58 @@ fn lower_insn_to_regs<'a>(ctx: Ctx<'a>, inst: IRInst) {
                 unimplemented!("unimplemented lowering for opcode {:?}", op);
             }
         }
-
+        Opcode::Fcopysign => {
+            let dst = output_to_reg(ctx, inst, 0);
+            let lhs = input_to_reg(ctx, inst, 0);
+            let rhs = input_to_reg(ctx, inst, 1);
+            let is64 = flt_ty_is_64(ty.unwrap());
+            if !is64 {
+                let w64 = 0x8000_0000;
+                let tregI64 = ctx.alloc_tmp(RegClass::I64, I32);
+                let tregV128 = ctx.alloc_tmp(RegClass::V128, F64);
+                let t2regV128 = ctx.alloc_tmp(RegClass::V128, F64);
+                let t3regV128 = ctx.alloc_tmp(RegClass::V128, F64);
+                ctx.emit(Inst::imm_r(true, w64, tregI64));
+                ctx.emit(Inst::xmm_mov_rm_r(
+                    SseOpcode::Movd,
+                    RegMem::reg(tregI64.to_reg()),
+                    tregV128,
+                ));
+                ctx.emit(Inst::xmm_mov_rm_r(
+                    SseOpcode::Movaps,
+                    RegMem::reg(tregV128.to_reg()),
+                    t2regV128,
+                ));
+                ctx.emit(Inst::xmm_rm_r(
+                    SseOpcode::Andnps,
+                    RegMem::reg(lhs),
+                    t2regV128,
+                ));
+                ctx.emit(Inst::xmm_mov_rm_r(
+                    SseOpcode::Movss,
+                    RegMem::reg(rhs),
+                    t3regV128,
+                ));
+                // TODO: Input register should be writable. How to do this without adding another temp.
```

Comment:
It is totally fine to use another temp, because at this level we're operating on virtual registers; it is the register allocator's duty to try to coalesce those virtual registers into real registers. So feel free to use as many temporaries as you'll need.